File: src/org/cerion/symcalc/expression/function/core/NumberQ.java
package org.cerion.symcalc.expression.function.core;

import org.cerion.symcalc.exception.ValidationException;
import org.cerion.symcalc.expression.BoolExpr;
import org.cerion.symcalc.expression.Expr;
import org.cerion.symcalc.expression.function.Function;
import org.cerion.symcalc.expression.function.FunctionExpr;

public class NumberQ extends FunctionExpr {

    public NumberQ(Expr... e) {
        super(Function.NUMBERQ, e);
    }

    @Override
    protected Expr evaluate() {
        return new BoolExpr(get(0).isNumber());
    }

    @Override
    public void validate() throws ValidationException {
        validateParameterCount(1);
    }
}

File: src/main/kotlin/function/core/NumberQ.kt
package org.cerion.symcalc.function.core

import org.cerion.symcalc.exception.ValidationException
import org.cerion.symcalc.expression.BoolExpr
import org.cerion.symcalc.expression.Expr
import org.cerion.symcalc.function.FunctionExpr
import org.cerion.symcalc.expression.number.NumberExpr

class NumberQ(vararg e: Expr) : FunctionExpr(*e) {

    override fun evaluate(): Expr {
        return BoolExpr(get(0) is NumberExpr)
    }

    @Throws(ValidationException::class)
    override fun validate() {
        validateParameterCount(1)
    }
}
